### PR TITLE
docs: revamp section introducing Zephyr

### PR DESCRIPTION
**Pull request by nashif**
Add more features and re-order sections slightly.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

### CODE REVIEW

File: doc/introduction/introducing_zephyr.rst [hunk 1]
```diff
@@ -32,53 +32,94 @@ Distinguishing Features
...
+   Information about the system is extracted from the compiled DTS and used to
+   create the application image.
 
 **Memory Protection**
    Implements configurable architecture-specific stack-overflow protection,
    kernel object and device driver permission tracking, and thread isolation
    with thread-level memory protection on x86, ARC, and ARM architectures,
    userspace, and memory domains.
 
+   For platforms without MMU/MPU and memory constrained devices support for
```
Comment (dbkinder) [inline doc/introduction/introducing_zephyr.rst]: Better would be:

    For platforms without MMU/MPU and memory constrained devices, supports
    combining ...

**Review by dbkinder — CHANGES_REQUESTED** [review]
one tweak, otherwise LGTM

**Review by dbkinder — APPROVED** [review]
thanks!

**Comment by carlescufi** [comment]
Nice! Thanks for this @nashif

**Pull request merged**